src/libcollections/slice.rs

```diff
@@ -1031,9 +1031,21 @@ pub trait SliceConcatExt<T: ?Sized> {
     /// # Examples
     ///
     /// ```
+    /// assert_eq!(["hello", "world"].join(" "), "hello world");
+    /// ```
+    #[stable(feature = "rust1", since = "1.0.0")]
+    fn join(&self, sep: &T) -> Self::Output;
+
+    /// Flattens a slice of `T` into a single value `Self::Output`, placing a
+    /// given separator between each.
+    ///
+    /// # Examples
+    ///
+    /// ```
     /// assert_eq!(["hello", "world"].connect(" "), "hello world");
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[deprecated(since = "1.3.0", reason = "renamed to join")]
     fn connect(&self, sep: &T) -> Self::Output;
```

Comment:
Oh yes. Sorry I saw that this PR was breaking all implementations anyway,
so I immediately paged out "having to implement join" as a breaking change.
:)

Longterm I think connect-defaults-to-join is the right one. Regardless,
sliceconcatext isn't really something that's meant to be impl'd... it's
just a hack we use internally, I think.

On Sat, Jul 11, 2015 at 10:45 AM, Eduard Burtescu <notifications@github.com>
wrote:

> In src/libcollections/slice.rs
> <https://github.com/rust-lang/rust/pull/26957#discussion_r34414816>:
>
> >      /// assert_eq!(["hello", "world"].connect(" "), "hello world");
> >      /// ```
> >      #[stable(feature = "rust1", since = "1.0.0")]
> > +    #[deprecated(since = "1.3.0", reason = "renamed to join")]
> >      fn connect(&self, sep: &T) -> Self::Output;
>
> Without a default on join, no user impls will compile after this change.
> But if the trait is unstable... It's probably fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/26957/files#r34414816>.
>